sdk/ai/azure-ai-projects/samples/agents/sample_agents_enterprise_code_interpretery.py

```diff
@@ -0,0 +1,113 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""
+FILE: sample_agents_enterprise_file_search.py
+
+DESCRIPTION:
+    This sample demonstrates how to add files to agent during the vector store creation.
+
+USAGE:
+    python sample_agents_enterprise_file_search.py
+
+    Before running the sample:
+
+    pip install azure.ai.projects azure-identity azure-ai-ml
+
+    Set this environment variables with your own values:
+    PROJECT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+"""
+
+import os
+from azure.ai.projects import AIProjectClient
+from azure.ai.projects.models import (
+    CodeInterpreterTool,
+    VectorStoreDataSource,
+    VectorStoreDataSourceAssetType,
+    CodeInterpreterToolResource,
+    ToolResources,
+)
+from azure.identity import DefaultAzureCredential
+
+from pathlib import Path
+
+# Create an Azure AI Client from a connection string, copied from your AI Studio project.
+# At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
+# Customer needs to login to Azure subscription via Azure CLI and set the environment variables
+
+credential = DefaultAzureCredential()
+project_client = AIProjectClient.from_connection_string(
+    credential=credential, conn_str=os.environ["PROJECT_CONNECTION_STRING"]
+)
+
+with project_client:
+
+    # upload a file and wait for it to be processed
+    # [START upload_file_and_create_agent_with_code_interpreter]
+    # We will upload the local file to Azure and will use it for vector store creation.
+    _, asset_uri = project_client.upload_file("./product_info_1.md")
+
+    # create a vector store with no file and wait for it to be processed
+    ds = VectorStoreDataSource(asset_identifier=asset_uri, asset_type=VectorStoreDataSourceAssetType.URI_ASSET)
+
+    # create code interpreter tool and resource
+    code_interpreter = CodeInterpreterTool()
+    code_interpreter_resource = CodeInterpreterToolResource(data_sources=[ds])
+    tool_resources = ToolResources(code_interpreter=code_interpreter_resource)
```

Comment:
I think if we have higher level CodeInterpreterTool, we shall enable all functionalities via that

Reply:
CodeInterpreterTool doesn't take dataSources now.   It should be an improvement for the future.   But for now to make this feature work, I need to use this approach.